compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -2940,20 +2940,22 @@ declare_lint! {
     ///
     /// ### Explanation
     ///
-    /// The example above shows how a user might accidentally cast a function pointer (rather
-    /// than an integer constant) to a non-`usize` integer type. Though this is currently permitted
-    /// by Rust, there is very little reason to cast a pointer to any integer other than a `usize`.
-    /// Therefore, any instances of such casts are likely to be bugs.
+    /// The example above shows how a user might accidentally cast a function pointer (rather than
+    /// an integer constant) to an integer type other than `usize` or `u64`. Though this is
+    /// currently permitted by Rust, there is very little reason to cast a pointer to any integer
+    /// other than a `usize` or `u64`. Therefore, any instances of such casts are likely to be
+    /// bugs.
     ///
     /// This lint warns against those cases.
     ///
     /// In the future, we may want to make this a hard error.
     ///
-    /// To cast a pointer to a non-`usize` integer without triggering the lint, you can first cast
-    /// to a `usize` and then to the integer type, e.g. `ptr as usize as u32`.
+    /// To cast a pointer to an integer type other than `usize` or `u64` without triggering the
+    /// lint, you can first cast to a `usize` or `u64` and then to the integer type, e.g. `ptr as
```

Comment:
I think it'd be fine to say "you can first cast to a `usize` and then" here; that will give people guidance if they don't already have a specific need.

The rest of the change seems fine.